### PR TITLE
Add Github Workflow for testing with docker-compose

### PR DESCRIPTION
**Pull request by skanderm**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Docker
```
Comment (paulcretu) [inline .github/workflows/test.yaml:1]:
```suggestion
name: CI
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (paulcretu) [inline .github/workflows/test.yaml:6]:
```suggestion
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

Shouldn't we run tests on commit/push as well?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (paulcretu) [inline .github/workflows/test.yaml:13]:
```suggestion
        uses: actions/checkout@v4
```
v4 is latest https://github.com/actions/checkout#checkout-v4

**Review by paulcretu — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ services:
       - REDIS_HOST=cache
       - REDIS_PORT=6379
     ports:
-      - "3000:3000"
+      # - "3000:3000"
```
[the remark this answers is not in the document]
Comment (skanderm) [reply docker-compose.yml:12]: Oh sure. I thought I ran into issues with having the port open, but testing it just now worked fine.

**Review by paulcretu — APPROVED** [review]

**Pull request merged**